plugins/modules/digital_ocean_firewall.py

```diff
@@ -314,11 +314,30 @@ def __init__(self, module):
         self.firewalls = self.get_firewalls()
 
     def get_firewalls(self):
-        base_url = self.baseurl + "?"
```

Comment:
The results from the api are integers, but what the user defines may be an array of strings (if I'm not mistaken), especially considering that jinja expressions are basically strings, and an integer value may become strings along the way. At least that's how I was able to simulate the error, so the change make the module more fault tolerant.

Regarding the quotes, it was in the function I changed so I didn't see a problem because there was no change in behaviour, unless the ansible team prefers double quotes instead of single quotes. In most cases I would use a linter with a formatter like pylance or pylint and not care about quotes (on save it would replace by the project defaults), but neither pylance nor pylint worked well and I saw no documentation at https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_general.html about a linter, so I disabled linter and auto-formatting.